bgls/simulator_test.py

```diff
@@ -352,3 +352,39 @@ def test_mps_results_match_state_vec():
     result_mps = sim_mps.run(circuit, repetitions=100)
 
     assert result_mps == result_state_vector
+
+
+def test_dense_moments_intermediate_measurement_counting():
```

Comment:
How about `test_intermediate_measurements_are_ignored`? I don't know what you mean by counting